pkg/test/framework/components/istio/util.go

```diff
@@ -69,11 +69,20 @@ func waitForValidationWebhook(ctx resource.Context, cluster resource.Cluster, cf
 }
 
 func (i *operatorComponent) RemoteDiscoveryAddressFor(cluster resource.Cluster) (net.TCPAddr, error) {
```

Comment:
While it makes sense to modify the logic of this function, I agree strongly that the ingress APIs (`ingressImpl.DiscoveryAddress`, `ingressImpl.getAddressInner`) shouldn't be changed because they're meant to get addresses for the ingress Service. Rather, the internals of `getAddressInner` can be moved to a helper function for getting either a nodeIP and port for a given Service or the LoadBalancer address and port. The helper can be used to get the address for istiod